### ISSUE TITLE
Use NavbarBrand component for Navbar brand

### ISSUE THREAD

**Issue by taion** [react-bootstrap/react-bootstrap#1158]
There's no real reason to have `brand` be a prop on `Navbar` - it's a sibling of the `Nav` rendered inside it, so it'd make just as much sense as a separate component. The new API should then be e.g.:

```js
<Navbar>
  <NavbarBrand>My Brand</NavbarBrand>
  <Nav>{/* ... */}</Nav>
</Navbar>
```

**Comment by kwnccc** [comment]
Hi @taion , I'm new to this repo. I'd love to start contributing to it. I was thinking of starting this implementation if there's no one else working on it.

**Comment by taion** [comment]
Nobody as far as I'm aware. Go for it!

**Comment by taion** [comment]
Was resolved by #1184. Thanks @kwnccc !
